### PR TITLE
Shovel support.

### PR DESCRIPTION
**Pull request by emersonkopp**
Added support to shovel management.

### CODE REVIEW

File: src/main/java/com/rabbitmq/http/client/domain/ShovelDetails.java [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+	public String getSourceQueue() {
+		return sourceQueue;
+	}
+
+	public void setSourceQueue(String sourceQueue) {
+		this.sourceQueue = sourceQueue;
+	}
+
+	public String getDestURI() {
```
Comment (michaelklishin) [inline src/main/java/com/rabbitmq/http/client/domain/ShovelDetails.java]: Thank you. Can we please use full words in field and accessor names? Yes, the API ones are abbreviated and can't change but we can do better here :)

Comment (emersonkopp) [inline src/main/java/com/rabbitmq/http/client/domain/ShovelDetails.java]: Done. Changed to getDestinationURI.

File: bin/before_build.sh [hunk 2]
```diff
@@ -12,4 +17,8 @@ $HOP_RABBITMQCTL set_permissions -p / guest ".*" ".*" ".*"
 $HOP_RABBITMQCTL eval 'supervisor2:terminate_child(rabbit_mgmt_sup_sup, rabbit_mgmt_sup), application:set_env(rabbitmq_management,       sample_retention_policies, [{global, [{605, 1}]}, {basic, [{605, 1}]}, {detailed, [{10, 1}]}]), rabbit_mgmt_sup_sup:start_child().'
 $HOP_RABBITMQCTL eval 'supervisor2:terminate_child(rabbit_mgmt_agent_sup_sup, rabbit_mgmt_agent_sup), application:set_env(rabbitmq_management_agent, sample_retention_policies, [{global, [{605, 1}]}, {basic, [{605, 1}]}, {detailed, [{10, 1}]}]), rabbit_mgmt_agent_sup_sup:start_child().'
 
+# Enable shovel plugin
+$PLUGINS enable rabbitmq_shovel
```
Comment (michaelklishin) [inline bin/before_build.sh]: `PLUGINS` is not defined. I will correct to `HOP_RABBITMQ_PLUGINS` before QA.

**Comment by michaelklishin** [comment]
Thank you! Heads-up: in RabbitMQ `3.7.0`, Shovel becomes multi-protocol so `ShovelInfo` will have to change. This means Hop will potentially need to have two support series :/

File: bin/before_build.sh (hunk 2 above)
Comment (emersonkopp) [inline bin/before_build.sh]: OK. Thank you.

**Pull request merged**